File: leetcode/leetcode_2215.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        unordered_set <int> s1(nums1.begin(),nums1.end());
        unordered_set <int> s2(nums2.begin(),nums2.end());

        vector <int> v1;
        vector <int> v2;

        for(auto n:s1) {
            if(s2.count(n)==0) {
                v1.push_back(n);
            }
        }

        for(auto n:s2) {
            if(s1.count(n)==0) {
                v2.push_back(n);
            }
        }

        return {v1,v2};        
    }
};

File: leetcode/leetcode_2215.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[][]}
 */
var findDifference = function(nums1, nums2) {

    let s1 = new Set(nums1);
    let s2 = new Set(nums2);

    for(let i of s1) {
        if(s2.has(i)) {
            s2.delete(i);
            s1.delete(i);
        }
    }

    return [Array.from(s1),Array.from(s2)];
    
};